core/common/src/main/java/tachyon/wire/BlockInfo.java

```diff
@@ -0,0 +1,140 @@
+/*
+ * Licensed to the University of California, Berkeley under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package tachyon.wire;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Objects;
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+
+/**
+ * The block descriptor.
+ */
+@NotThreadSafe
+public final class BlockInfo implements WireType<tachyon.thrift.BlockInfo> {
+  private long mBlockId;
+  private long mLength;
+  private List<BlockLocation> mLocations = Lists.newArrayList();
+
+  /**
+   * Creates a new instance of {@link BlockInfo}.
+   */
+  public BlockInfo() {}
```

Comment:
Got it. Ideally I think we should only be able to derive this object from the underlying object, but I think the code was not written in such a way before.